include/hjson/hjson.h

```diff
@@ -142,6 +146,14 @@ class Value {
   // Throws if used on VECTOR or MAP
   double to_double() const;
   std::string to_string() const;
+
+  // commentary access functions
+  std::string& comment_pre();
+  std::string& comment_post();
+  std::string& comment_inside();
```

Comment:
Are these setter-functions? I would prefer `set_comment_inside(const std::string&)` etc instead.

Reply:
Most of my parsing builds onto having a reference to the comment-string which should be filled. See `_readComment` for example. Of course one could build a temporary string and copy / move it later, but it's a little slimmer (performance & code) this way. And one can use a call like `val.comment_post() += "my addition to comment";`.

But if you want to get rid of the refrence return value, one should probably use `set_comment_inside(std::string)` instead and internally move the parameter. See also [here](https://www.codesynthesis.com/~boris/blog/2012/06/26/efficient-argument-passing-cxx11-part2/): "So, in summary, pass by value is optimal if we know for sure we are going to copy the argument."

We have a trade-off between performance & usability one one side and safety on the other. I think you should decide...

Reply:
Thanks for the link! `set_comment_inside(std::string)` looks good. I prefer safety over performance for this library.